core/src/main/java/com/sequenceiq/cloudbreak/service/proxy/ProxyConfigService.java

```diff
@@ -0,0 +1,117 @@
+
+    @Inject
+    private ProxyConfigRepository proxyConfigRepository;
+
+    @Inject
+    private ClusterRepository clusterRepository;
+
+    @Inject
+    private AuthorizationService authorizationService;
+
+    public Set<ProxyConfig> retrievePrivateProxyConfigs(IdentityUser user) {
+        return proxyConfigRepository.findForUser(user.getUserId());
+    }
+
+    public ProxyConfig getPrivateProxyConfig(String name, IdentityUser user) {
+        ProxyConfig proxyConfig = proxyConfigRepository.findByNameInUser(name, user.getUserId());
+        if (proxyConfig == null) {
+            throw new NotFoundException(String.format("Proxy configuration '%s' not found.", name));
+        }
+        return proxyConfig;
+    }
+
+    public ProxyConfig getPublicProxyConfig(String name, IdentityUser user) {
+        ProxyConfig proxyConfig = proxyConfigRepository.findOneByName(name, user.getAccount());
+        if (proxyConfig == null) {
+            throw new NotFoundException(String.format("Proxy configuration '%s' not found.", name));
+        }
+        return proxyConfig;
+    }
+
+    public Set<ProxyConfig> retrieveAccountProxyConfigs(IdentityUser user) {
+        return user.getRoles().contains(IdentityUserRole.ADMIN) ? proxyConfigRepository.findAllBasedOnAccount(user.getAccount())
+                : proxyConfigRepository.findPublicInAccountForUser(user.getUserId(), user.getAccount());
+    }
+
+    public ProxyConfig get(Long id) {
+        ProxyConfig proxyConfig = proxyConfigRepository.findById(id);
+        if (proxyConfig == null) {
+            throw new NotFoundException(String.format("Proxy configuration '%s' not found.", id));
+        }
+        authorizationService.hasReadPermission(proxyConfig);
+        return proxyConfig;
+    }
+
+    public void delete(Long id, IdentityUser user) {
+        ProxyConfig proxyConfig = proxyConfigRepository.findByIdInAccount(id, user.getAccount());
+        if (proxyConfig == null) {
+            throw new NotFoundException(String.format("Proxy configuration '%s' not found.", id));
+        }
+        delete(proxyConfig);
+    }
+
+    public void delete(String name, IdentityUser user) {
+        ProxyConfig proxyConfig = proxyConfigRepository.findByNameBasedOnAccount(name, user.getAccount(), user.getUserId());
+        if (proxyConfig == null) {
+            throw new NotFoundException(String.format("Proxy configuration '%s' not found.", name));
+        }
+        delete(proxyConfig);
+    }
+
+    public ProxyConfig create(IdentityUser user, ProxyConfig proxyConfig) {
+        LOGGER.debug("Creating Proxy configuration: [User: '{}', Account: '{}']", user.getUsername(), user.getAccount());
+        proxyConfig.setOwner(user.getUserId());
+        proxyConfig.setAccount(user.getAccount());
+        return proxyConfigRepository.save(proxyConfig);
+    }
+
+    public ProxyConfig create(ProxyConfig proxyConfig) {
```

Comment:
Is it possible to publish only create(IdentityUser, ProxyConfig), so we don't need this method and we doon't need the validations as well.